### PR TITLE
Switch from travis to GitHub actions

### PR DESCRIPTION
**Pull request by jamesls**
There's few housekeeping tasks I'd like to do with this repo but we need CI in place before making those changes.

This pulls in #60 but updates it a bit to start with a simpler base set of recent go versions.  Not opposed to adding more versions later.

There's a couple of follow up tasks from this:

* Remove internal vendor of testify (not needed)
* Fix linting errors so we can renable `make check` in CI.

cc @skmcgrail

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,6 +45,5 @@ bench:
 pprof-cpu:
 	go tool pprof ./go-jmespath.test ./cpu.out
 
-test-internal-testify:
-	cd internal/testify && go test ./...
-
+install-golint:
+	go install golang.org/x/lint/golint@latest
```
Comment (skmcgrail) [inline Makefile:49]: Technically this modifies your go.mod file in the project to add this dependency.

`go run` can be used to invoke commands without actually having to install these explicitly.

So for example you can just do `go run golang.org/x/lint/golint <arguments>` wherever you need to run go linting. You can see in our V2 SDK repo we use this pattern to utilize some of our repository release tooling without actually having it modify our dependency descriptor: https://github.com/aws/aws-sdk-go-v2/blob/main/Makefile#L424

**Review by skmcgrail — APPROVED** [review]
Minor suggestions

File: Makefile (hunk 1 above)
Comment (jamesls) [reply Makefile:49]: Np I'll update, just want to make sure I understand the point about go.mod (trying to catch up with this module stuff), are you talking about the go.mod in this repo? When I run `go install ...` I see that it adds the binary to the global `~/go/bin/` dir, but it's not modifying any files in this repo.

```
/tmp/testitout/go-jmespath (sync-tests u=) $ git status
On branch sync-tests
Your branch is up to date with 'origin/sync-tests'.

nothing to commit, working tree clean
/tmp/testitout/go-jmespath (sync-tests u=) $ golint
bash: golint: command not found
/tmp/testitout/go-jmespath (sync-tests u=) $ go install golang.org/x/lint/golint@latest
/tmp/testitout/go-jmespath (sync-tests u=) $ golint -h
Usage of golint:
	golint [flags] # runs on package in current directory
	
      ...

/tmp/testitout/go-jmespath (sync-tests u=) $ which golint
/Users/j/go/bin/golint
/tmp/testitout/go-jmespath (sync-tests u=) $ git status
On branch sync-tests
Your branch is up to date with 'origin/sync-tests'.

nothing to commit, working tree clean
/tmp/testitout/go-jmespath (sync-tests u=) $ cat go.mod
module github.com/jmespath/go-jmespath

go 1.14

require github.com/jmespath/go-jmespath/internal/testify v1.5.1
```

Comment (skmcgrail) [reply Makefile:49]: Oh never mind, I was incorrect, I thought it did for some reason. You are good then 👍

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          - 1.16
+          - 1.15
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Install deps 
+        run: go get -d ./...
```
Comment (skmcgrail) [inline .github/workflows/ci.yml:24]: I thought I commented on this in my previous review, but I guess not. Suggestion would be to make this `go mod download`

**Pull request merged**